go/host/enclave/guardian.go

```diff
@@ -517,81 +519,50 @@ func (g *Guardian) periodicBatchProduction() {
 	}
 }
 
-func (g *Guardian) periodicRollupProduction() { //nolint:gocognit
+func (g *Guardian) periodicRollupProduction() {
 	defer g.logger.Info("Stopping rollup production")
 
-	// rollup at least at every g.rollupInterval
-	rollupInterval := g.rollupInterval
-	if rollupInterval == 0 {
-		g.logger.Crit("invalid rollup interval 0")
-	}
-	rollupIntervalBasedTicker := time.NewTicker(rollupInterval)
-
-	// every 5sec check if it should create a new rollup
-	rollupSizeCheckTicker := time.NewTicker(5 * time.Second)
+	// check rollup every l1 block time
+	rollupCheckTicker := time.NewTicker(g.blockTime)
+	lastSuccessfulRollup := time.Now()
 
 	for {
 		select {
-		case <-rollupIntervalBasedTicker.C:
-			if !g.state.IsUpToDate() {
-				// if we're behind the L1, we don't want to produce rollups
-				g.logger.Debug("skipping rollup production because L1 is not up to date", "state", g.state)
-				continue
-			}
-
-			fromBatch, err := g.getLastestBatchNo()
-			if err != nil {
-				g.logger.Warn("encountered error while trying to retrieve latest sequence number", log.ErrKey, err)
-				continue
-			}
-
-			producedRollup, err := g.enclaveClient.CreateRollup(fromBatch)
-			if err != nil {
-				g.logger.Error("unable to produce rollup", log.ErrKey, err)
-			} else {
-				fmt.Println(time.Now().String(), " - PRODUCING rollup from config'd rollupInterval - lastBatchNo: ", fromBatch)
-				g.sl.L1Publisher().PublishRollup(producedRollup)
-				fmt.Println(time.Now().String(), " - FINISHED PRODUCED rollup from config'd rollupInterval ")
-			}
-
-		case <-rollupSizeCheckTicker.C:
+		case <-rollupCheckTicker.C:
 			if !g.state.IsUpToDate() {
 				// if we're behind the L1, we don't want to produce rollups
 				g.logger.Debug("skipping rollup production because L1 is not up to date", "state", g.state)
 				continue
 			}
 
-			fromBatch, err := g.getLastestBatchNo()
+			fromBatch, err := g.getLatestBatchNo()
```

Comment:
I think we should have another skip here that's like:
```
if time.Since(lastSuccessfulRollup) < minRollupInterval {
   continue // only just produced a rollup, allow time for enclave to consume
}
```
where minRollupInterval can probably just be like 1*blocktime (it's mostly just incase the timer fires soon after the previous run, not sure how the timer works and if that's possible but I assume it is)

Reply:
Yeah, I like to have a minRollupInterval as a performance improvement, perhaps at a later stage?
Mainly due to the not being clear what the minRollupInterval should be.

Should be ok to not have this check for now because:
- This is an awaited method. Once the rollup is created, it will wait for the receipt, which guarantees one block spacing.
- This whole process then is called after the 1*l1blocktime duration which is AFTER the success tx mint. So after the await of the receipt (step above) it waits again another block to check if it should rollup.

Would it make sense to add a few timers/logs so that it's possible to monitor this behavior? (for future improvements?)

Reply:
Could be slightly confusing when the timer fires straightaway but shouldn't be a problem, like you say probably just optimisation at worst.

Reply:
Yeah as discussed on discord the race condition I was worried about isn't an issue any more because you're using the L1 data directly to decide how far we have rolled up. Disregard this!